netbench/netbench-driver/src/bin/netbench-driver-s2n-tls-client.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use netbench::{multiplex, scenario, Result};
+use netbench_driver::Allocator;
+use s2n_tls::raw::{
+    config::{Builder, Config},
+    error::Error,
+    security::DEFAULT_TLS13,
+};
+use s2n_tls_tokio::{TlsConnector, TlsStream};
+use std::{collections::HashSet, future::Future, net::SocketAddr, pin::Pin, sync::Arc};
+use structopt::StructOpt;
+use tokio::net::TcpStream;
+
+#[global_allocator]
+static ALLOCATOR: Allocator = Allocator::new();
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> Result<()> {
+    Client::from_args().run().await
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Client {
+    #[structopt(flatten)]
+    opts: netbench_driver::Client,
+}
+
+impl Client {
+    pub async fn run(&self) -> Result<()> {
+        let addresses = self.opts.address_map().await?;
+        let scenario = self.opts.scenario();
+
+        let client = self.client()?;
+        let client = netbench::Client::new(client, &scenario, &addresses);
+        let mut trace = self.opts.trace();
+        let mut checkpoints = HashSet::new();
+        let mut timer = netbench::timer::Tokio::default();
+        client.run(&mut trace, &mut checkpoints, &mut timer).await?;
+
+        Ok(())
+    }
+
+    fn client(&self) -> Result<ClientImpl> {
+        let connector = TlsConnector::new(self.config()?.build()?);
+        let connector: s2n_tls_tokio::TlsConnector<Config> = connector.into();
+        let connector = Arc::new(connector);
+
+        let config = multiplex::Config::default();
+
+        Ok(ClientImpl {
+            config,
+            connector,
+            id: 0,
+        })
+    }
+
+    fn config(&self) -> Result<Builder, Error> {
+        let mut builder = Config::builder();
+        builder.set_security_policy(&DEFAULT_TLS13)?;
+        for ca in self.opts.certificate_authorities() {
+            builder.trust_pem(ca.pem.as_bytes())?;
+        }
+        unsafe {
+            builder.disable_x509_verification()?;
```

Comment:
We shouldn't need this if everything is configured correctly. And it's technically cheating a little bit on performance to disable it.

Reply:
Fixed: 1e861d47e50afdd01da93b1cfc94f41af8b1fec4